crates/rolldown/src/stages/generate_stage/render_chunk_to_assets.rs

```diff
@@ -163,18 +165,49 @@ impl<'a> GenerateStage<'a> {
       index_vec![IndexSet::default(); chunk_graph.chunks.len()];
     let mut index_preliminary_assets: IndexInstantiatedChunks =
       IndexVec::with_capacity(chunk_graph.chunks.len());
-    try_join_all(chunk_graph.chunks.iter_enumerated().map(|(chunk_idx, chunk)| async move {
-      let mut ctx = GenerateContext {
-        chunk_idx,
-        chunk,
-        options: self.options,
-        link_output: self.link_output,
-        chunk_graph,
-        plugin_driver: self.plugin_driver,
-        warnings: vec![],
-      };
-      EcmaGenerator::instantiate_chunk(&mut ctx).await
-    }))
+    let chunk_to_codegen_ret = chunk_graph
+      .chunks
+      .raw
+      .par_iter()
+      .map(|item| {
+        item
+          .modules
+          .par_iter()
```

Comment:
2 `par_iters` is really costly, try flatten them first in a single thread and then `part_iter` it?

Spawning threads is really slow :-/

Reply:
rayon will reuse the global thread pool,  the real cost part is sent to the work queue(it is trivial compared to ast codegen). I don't flatten them because you need to group by them whatever at the end.